pkgs/development/libraries/libxcrypt/default.nix

```
@@ -0,0 +1,32 @@
+{ stdenv, fetchFromGitHub, autoconf, automake, libtool, pkg-config }:
+
+stdenv.mkDerivation rec {
+  pname = "libxcrypt";
+  version = "4.4.16";
+
+  src = fetchFromGitHub {
+    owner = "besser82";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256:1yyn33h9f0w4aahknvpqmzxjxzfhqy6g1d8x6x9wlwbpazx98xx7";
+  };
+
+  preConfigure = ''
+    patchShebangs autogen.sh
+    ./autogen.sh
+  '';
+
+  configureFlags = stdenv.lib.optional stdenv.hostPlatform.isx86 "--enable-fat=yes";
```

Personal taste of mine:

```suggestion
  configureFlags = [
  ]
    ++ stdenv.lib.optionals stdenv.hostPlatform.isx86 [ "--enable-fat=yes" ]
  ;
```

This makes future diffs cleaner.